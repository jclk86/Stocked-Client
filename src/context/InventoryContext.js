import React, { Component } from "react";

const InventoryContext = React.createContext({
  inventoryList: [],
  tagsList: [],
  error: null,
  addInventoryItem: () => {},
  setInventoryList: () => {},
  updateInventoryItem: () => {},
  deleteInventoryItem: () => {},
  clearInventoryItem: () => {},
  setError: () => {},
  clearError: () => {}
});
export default InventoryContext;

export class InventoryProvider extends Component {
  state = {
    inventoryList: [],
    tagsList: [],
    error: null
  };

  setInventoryList = inventoryList => {
    this.setState({ inventoryList });
  };

  setTagsList = tagsList => {
    this.setState({ tagsList });
  };

  setError = error => {
    console.error(error);
    this.setState({ error }); // these must be set with conditionals. on the component rendering area you want to render error message if not rendered. Context becomes errorboundary, Not need fo errorBoundary component
  };

  clearError = () => {
    this.setState({ error: null });
  };

  addInventoryItem = item => {
    this.setState({ inventoryList: [...this.state.inventoryList, item] });
  };

  updateInventoryItem = updatedItem => {
    this.setState({
      inventoryList: this.state.inventoryList.map(item =>
        item.item_id !== updatedItem.item_id ? item : updatedItem
      )
    });
  };

  deleteInventoryItem = deletedItemId => {
    this.setState({
      inventoryList: this.state.inventoryList.filter(
        item => item.itemId !== parseInt(deletedItemId)
      )
    });
  };

  // clearItem = () => {
  //   this.setInventoryList(nullInventory)
  //   this.setComments([])
  // }

  render() {
    const contextValue = {
      inventoryList: this.state.inventoryList,
      user_id: 1, // hardcoded. Change
      tagsList: this.state.tagsList,
      setInventoryList: this.setInventoryList,
      setTagsList: this.setTagsList,
      addInventoryItem: this.addInventoryItem,
      updateInventoryItem: this.updateInventoryItem,
      deleteInventoryItem: this.deleteInventoryItem
    };
    return (
      <InventoryContext.Provider value={contextValue}>
        {this.props.children}
      </InventoryContext.Provider>
    );
  }
}
