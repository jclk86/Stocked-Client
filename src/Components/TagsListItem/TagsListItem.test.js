import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import TagListItem from "./TagsListItem";
import InventoryContext from "../../context/InventoryContext";

describe(`TagsListItem Component`, () => {
  const tag = {
    tagId: 1
  };

  it("renders tags without issues", () => {
    const wrapper = shallow(<TagListItem tag={tag}></TagListItem>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
