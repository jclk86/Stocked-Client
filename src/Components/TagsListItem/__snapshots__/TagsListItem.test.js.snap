// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagsListItem Component renders tags without issues 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
